01-syntax/03-data-structures/deep-comparation/deepEqual.js

```diff
@@ -13,21 +11,217 @@ var obj = {
   object: 2
 };
 
+function deepEqual(firstObj, secondObj) {
+  var firstKey;
+  var secondKey;
+  var same;
+
+  function isObject(statement) {
+    return (statement && typeof statement === 'object');
+  }
+
+  arguments[0] = firstObj || {};
```

Comment:
I won't suggest to use such a way of providing defaults for argument's 
better to explicitly define new variables in your function fro example 

 ```js

function deepEqual(obj1, obj2){

var firstObj =  obj1 || {};
var secondObj = obj2 || {};
//....
}

```